commands/bnet.js

```
@@ -0,0 +1,443 @@
+           Array.from(this.battleTags).sort().join(', ');
+  }
+}
+
+/**
+ * Retrieves the specified channel using the specified client.
+ * @param {Discord.Client} client The client to use for querying
+ * @param {string} channelId The snowflake for the channel to retrieve
+ * @return {Discord.Channel} The channel retrieve, or null if it does not
+ * exist.
+ */
+function getChannelById(client, channelId) {
+  if (!client) {
+    console.error('Error getting channel: No cilent specified');
+    return null;
+  }
+
+  if (!channelId) {
+    console.error('Error getting channel: No channel ID specified');
+    return null;
+  }
+
+  const channel = client.channels.get(channelId);
+  if (!channel) {
+    console.error(`Error getting channel: Could not find channel with ` +
+                  `ID ${channelId}`);
+  }
+  return channel;
+};
+
+/**
+ * Adds all BattleTags in the BattleTags list to the entry for the specified
+ * user or text key. User keys are their Snowflake ID. If an entry was storing
+ * a text ID and a user is specified the text ID is overriden and the list of
+ * BattleTags is reset.
+ * @param {string} key The key of the user to add too
+ * @param {(Discord.GuildMember|string)} userOrTextId The GuildMember or text ID
+ * to store
+ * @param {string[]} battleTagList A list of BattleTags to add
+ */
+function addBattleTags(key, userOrTextId, battleTagList) {
+  let discordUser = null;
+  let textId = null;
+  if (userOrTextId instanceof Discord.GuildMember) {
+    discordUser = userOrTextId;
+  } else {
+    textId = userOrTextId;
+  }
+  const debugString = discordUser ? discordUser.toString() : 'null';
+  console.log(`Add BattleTag: ${key}, ${debugString}, ${textId}: ` +
+              `${battleTagList}`);
+  if (!battleTags.has(key)) {
+    battleTags.set(key, new BattleTagEntry(discordUser, textId, battleTagList));
+  } else {
+    const existingEntry = battleTags.get(key);
+    if (!discordUser && existingEntry.user) {
```

Do we just silently fail if a user tries to add a text id but the tagentry uses a guildmember?

---

No longer an issue as this has been removed